.github/workflows/benchmarks-jit-on-off.yml

```diff
@@ -0,0 +1,182 @@
+          echo "- Commit: \`${{ github.sha }}\`" >> results.md
+          echo "- Paths: \`${{ env.BENCH_PATHS }}\`" >> results.md
+          echo "- Runs: \`${{ env.BENCH_RUNS }}\` (after warmup)" >> results.md
+          echo "" >> results.md
+
+          echo "### Raw Statistics (${BENCH_RUNS} runs each)" >> results.md
+          echo "" >> results.md
+          echo "| Branch | Min | Max | Mean | StdDev | Median |" >> results.md
+          echo "|--------|-----|-----|------|--------|--------|" >> results.md
+
+          echo "**JIT OFF:**"
+          OFF_STATS=$(awk '$2 == "AAD" { print $4 }' build-off/samples/LiborSwaptionPricer/benchmark_off.log | datamash min 1 max 1 mean 1 sstdev 1 median 1)
+          OFF_TIME=$(echo "$OFF_STATS" | awk '{print $5}')
+          echo "| JIT OFF | $(echo "$OFF_STATS" | awk '{printf "%.3fs | %.3fs | %.3fs | %.3fs | %.3fs", $1, $2, $3, $4, $5}') |" >> results.md
+
+          echo "**JIT ON:**"
+          ON_STATS=$(awk '$2 == "AAD" { print $4 }' build-on/samples/LiborSwaptionPricer/benchmark_on.log | datamash min 1 max 1 mean 1 sstdev 1 median 1)
+          ON_TIME=$(echo "$ON_STATS" | awk '{print $5}')
+          echo "| JIT ON | $(echo "$ON_STATS" | awk '{printf "%.3fs | %.3fs | %.3fs | %.3fs | %.3fs", $1, $2, $3, $4, $5}') |" >> results.md
+
+          echo "" >> results.md
+          echo "### Summary" >> results.md
+          echo "" >> results.md
+
+          DIFFERENCE=$(echo "scale=6; $OFF_TIME - $ON_TIME" | bc | awk '{printf "%.3f", $1}')
+          PERCENTAGE=$(echo "scale=6; (($OFF_TIME - $ON_TIME) / $OFF_TIME) * 100.0" | bc | awk '{printf "%.2f", $1}')
+
+          echo "| Metric | Value |" >> results.md
+          echo "|--------|-------|" >> results.md
+          echo "| JIT OFF median | ${OFF_TIME}s |" >> results.md
+          echo "| JIT ON median | ${ON_TIME}s |" >> results.md
+          echo "| Difference | ${DIFFERENCE}s |" >> results.md
+          echo "| Change | ${PERCENTAGE}% |" >> results.md
+          echo "" >> results.md
+
+          # Determine if improvement or regression
+          if (( $(echo "$PERCENTAGE > 1" | bc -l) )); then
+            echo "> **Improvement**: JIT ON is ${PERCENTAGE}% faster than JIT OFF" >> results.md
+          elif (( $(echo "$PERCENTAGE < -1" | bc -l) )); then
+            echo "> **Regression**: JIT ON is $(echo "$PERCENTAGE * -1" | bc)% slower than JIT OFF" >> results.md
+          else
+            echo "> **No significant change**: Within 1% margin" >> results.md
+          fi
+
+          cat results.md
+
+      - name: Upload Results
+        uses: actions/upload-artifact@v5
+        with:
+          name: benchmark-results
+          path: results.md
+
+      - name: Post PR Comment
+        if: github.event_name == 'pull_request'
+        uses: peter-evans/create-or-update-comment@v5
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body-path: results.md
+          edit-mode: replace
```

Comment:
Did you try this in your fork before, if a PR created from a fork is actually allowed to post comments on PRs in this repository? 

Previously we had do this from a separate workflow that runs within this repository after the PR one, which gets the data and then posts the comment. See here for an example: https://github.com/auto-differentiation/xad/blob/main/.github/workflows/testresults.yml - the test results are pulled from the branch / PR build and posted from `main`.